### PR TITLE
Wallet DB Index changes

### PR DESCRIPTION
**Pull request by neurosis69**
Checked the wallet db for current indexes and suggest the following changes:

- added information for unused indexes and prepared drop stmt
- removed duplicate index on column created_at_time
- fixed probable wrong indexed columns (confirmed_at_index, used)
- fixed probably copy/paste error on delete for pool_state_transitions
- fixed multiple use of same index name (wallet_id, wallet_type)
- on action_queue table: only renamed the indexes , although the table is empty on my side and I see no actual query for index usage (maybe drop them?). Only done column is used in where clause.

### CODE REVIEW

**Comment by mariano54** [comment]
Can you remove the `.github` file changes?

File: chia/wallet/trading/trade_store.py [hunk 1]
```diff
@@ -205,7 +209,7 @@ async def get_not_sent(self) -> List[TradeRecord]:
         """
 
         cursor = await self.db_connection.execute(
-            "SELECT * from trade_records WHERE sent<? and confirmed=?",
```
Comment (arvidn) [inline chia/wallet/trading/trade_store.py:208]: is there no test covering this?
If not, we really need one (clearly)

File: chia/wallet/trading/trade_store.py [hunk 2]
```diff
@@ -225,7 +229,7 @@ async def get_all_unconfirmed(self) -> List[TradeRecord]:
         Returns the list of all trades that have not yet been confirmed.
         """
 
-        cursor = await self.db_connection.execute("SELECT * from trade_records WHERE confirmed=?", (0,))
```
Comment (arvidn) [inline chia/wallet/trading/trade_store.py:228]: and a test for this too

File: chia/wallet/wallet_action_store.py [hunk 3]
```diff
@@ -37,11 +37,11 @@ async def create(cls, db_wrapper: DBWrapper):
             )
         )
 
-        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS name on action_queue(name)")
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS aq_name on action_queue(name)")
```
Comment (arvidn) [inline chia/wallet/wallet_action_store.py:40]: why do you rename this?
It means everyone will need to spend a bunch of time at startup (when upgrading) to create a new index.

File: chia/wallet/wallet_action_store.py [hunk 4]
```diff
@@ -37,11 +37,11 @@ async def create(cls, db_wrapper: DBWrapper):
...
 
-        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS wallet_id on action_queue(wallet_id)")
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS aq_wallet_id on action_queue(wallet_id)")
 
-        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS wallet_type on action_queue(wallet_type)")
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS aq_wallet_type on action_queue(wallet_type)")
```
Comment (arvidn) [inline chia/wallet/wallet_action_store.py:44]: I expect a strong rationale for all of these name changes. Without dropping the previous indices, you also make the database larger (for not apparent reason)

File: chia/wallet/wallet_transaction_store.py [hunk 5]
```diff
@@ -52,25 +52,23 @@ async def create(cls, db_wrapper: DBWrapper):
             "CREATE INDEX IF NOT EXISTS tx_confirmed_index on transaction_record(confirmed_at_height)"
         )
 
-        await self.db_connection.execute(
-            "CREATE INDEX IF NOT EXISTS tx_created_index on transaction_record(created_at_time)"
-        )
-
+        # index tx_created_index is a duplicate to tx_created_time and can be dropped
+        await self.db_connection.execute("DROP INDEX IF EXISTS tx_created_index on transaction_record(created_at_time)")
```
Comment (arvidn) [inline chia/wallet/wallet_transaction_store.py:56]: is there a good reason to change the formatting here? I would expect `flake8` or `black` to complain

File: chia/wallet/wallet_transaction_store.py [hunk 6]
```diff
@@ -52,25 +52,23 @@ async def create(cls, db_wrapper: DBWrapper):
...
 
-        await self.db_connection.execute(
-            "CREATE INDEX IF NOT EXISTS tx_to_puzzle_hash on transaction_record(to_puzzle_hash)"
-        )
+        # this index is not used by any queries, don't create it for new
+        # installs, and remove it from existing installs in the future
+        # await self.db.execute("DROP INDEX IF EXISTS tx_to_puzzle_hash on transaction_record(to_puzzle_hash)")
 
-        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS wallet_id on transaction_record(wallet_id)")
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS tx_wallet_id on transaction_record(wallet_id)")
```
Comment (arvidn) [inline chia/wallet/wallet_transaction_store.py:71]: why do you rename this index? It would make everyone upgrading spending time creating a new (identical) index.

File: tests/testconfig.py [hunk 7]
```diff
@@ -2,7 +2,7 @@
 oses = ["ubuntu", "macos"]
 
 # Defaults are conservative.
-parallel = False
+parallel = True
```
Comment (arvidn) [inline tests/testconfig.py:5]: This looks like a risky change. I would put this in a separate PR

File: .github/workflows/build-test-macos-blockchain.yml [hunk 8]
```diff
@@ -100,7 +100,7 @@ jobs:
     - name: Test blockchain code with pytest
       run: |
         . ./activate
-        ./venv/bin/py.test tests/blockchain/test_*.py -s -v --durations 0
+        ./venv/bin/py.test tests/blockchain/test_*.py -s -v --durations 0 -n auto
```
Comment (arvidn) [inline .github/workflows/build-test-macos-blockchain.yml:103]: these files are auto generated. I don't immediately see what you may have changed causing them to be generated differently. maybe you can rebase on top of main.

**Comment by arvidn** [comment]
I would recommend you install `pre-commit` in your local repository. `pip install pre-commit` and `pre-commit install` (iirc)

**Comment by neurosis69** [comment]
> 
> 
> Can you remove the `.github` file changes?

my bad, I don't know what went wrong here. will remove them

File: chia/wallet/wallet_action_store.py (hunk 3 above)
Comment (neurosis69) [reply chia/wallet/wallet_action_store.py:40]: All wallet dbs synced on my side are missing indexes on action_queue table.
Reason is, that the index name "name" was used multiple times for different tables in the scripts.
For instance, the here mentioned index "name" is on my local db already created on table "users_wallets".
I'd expect that all wallet dbs are missing these 3 indexes on action_queue table and therefore it should be safe to rename it.

File: chia/wallet/wallet_action_store.py (hunk 4 above)
Comment (neurosis69) [reply chia/wallet/wallet_action_store.py:44]: same reason. Index named "wallet_type" was already created on table coin_record.
I'd recommend to use table acronym in index name or something similar so such problems don't happen.

File: chia/wallet/wallet_transaction_store.py (hunk 5 above)
Comment (neurosis69) [reply chia/wallet/wallet_transaction_store.py:56]: didn't get any `flake8` or `black` error on this. 
I can replace the CREATE INDEX by the DROP INDEX as well. 
I just added a comment and wanted the index to be dropped, as it is a 1:1 duplicate to tx_created_time.

File: chia/wallet/wallet_transaction_store.py (hunk 6 above)
Comment (neurosis69) [reply chia/wallet/wallet_transaction_store.py:71]: Similar to other indexes I renamed. On my wallet dbs there is currently no index on transaction_record table, as they are conflicting by name. index named "wallet_id" for example already exists on coin_record table.

File: tests/testconfig.py (hunk 7 above)
Comment (neurosis69) [reply tests/testconfig.py]: My bad, I didn't want this to be part of this PR. I'll remove it.

File: .github/workflows/build-test-macos-blockchain.yml (hunk 8 above)
Comment (neurosis69) [reply .github/workflows/build-test-macos-blockchain.yml]: Same here, was added by mistake. removed it.

**Comment by neurosis69** [comment]
> 
> 
> I would recommend you install `pre-commit` in your local repository. `pip install pre-commit` and `pre-commit install` (iirc)

tbh, I have installed and configured it according to guides you shared, but it obviously doesn't work like it should.
I clearly have some misconfiguration. Any guides I can check or more detailed how-tos you'd recommend on this?

File: chia/wallet/wallet_action_store.py (hunk 3 above)
Comment (arvidn) [reply chia/wallet/wallet_action_store.py:40]: I see. So, there is a bug where we attempt to create different indices with the same name, right?

Comment (arvidn) [reply chia/wallet/wallet_action_store.py:40]: or rather, several bugs

File: chia/wallet/wallet_user_store.py [hunk 9]
```diff
@@ -33,11 +33,17 @@ async def create(cls, db_wrapper: DBWrapper):
...
 
-        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS type on users_wallets(wallet_type)")
+        # this index is not used by any queries, don't create it for new
+        # installs, and remove it from existing installs in the future
+        # await self.db.execute("DROP INDEX IF EXISTS type on users_wallet(wallet_type)")
 
-        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS data on users_wallets(data)")
+        # this index is not used by any queries, don't create it for new
+        # installs, and remove it from existing installs in the future
+        # await self.db.execute("DROP INDEX IF EXISTS data on users_wallet(data)")
```
Comment (arvidn) [inline chia/wallet/wallet_user_store.py:46]: it it's not actually used anywhere, I don't think there's any backwards-compatibility reason to keep it. We should probably just drop these indices here

**Comment by arvidn** [comment]
sorry for taking so long understanding what the underlying problems were. As a general tip, it's usually better to submit separate PRs for separate fixes. That way, smaller parts can be reviewed and land at a time. I think the majority of this patch is ready to land for example.

I don't think we need to delay dropping the indices that aren't used anywhere though.

File: chia/wallet/wallet_action_store.py (hunk 3 above)
Comment (neurosis69) [reply chia/wallet/wallet_action_store.py:40]: Yes, this happened several times in different scripts. This PR resolves all such create index collisions I found.

File: chia/wallet/wallet_user_store.py (hunk 9 above)
Comment (neurosis69) [reply chia/wallet/wallet_user_store.py:46]: will delete these

File: chia/wallet/wallet_action_store.py (hunk 3 above)
Comment (neurosis69) [reply chia/wallet/wallet_action_store.py:40]: Yes, this happened several times in different scripts. This PR covers and fixes all such create index collisions I got aware of.

File: chia/wallet/wallet_user_store.py (hunk 9 above)
Comment (neurosis69) [reply chia/wallet/wallet_user_store.py:46]: I'll remove them.

**Comment by neurosis69** [comment]
> 
> 
> sorry for taking so long understanding what the underlying problems were. As a general tip, it's usually better to submit separate PRs for separate fixes. That way, smaller parts can be reviewed and land at a time. I think the majority of this patch is ready to land for example.
> 
> I don't think we need to delay dropping the indices that aren't used anywhere though.

I'll consider your suggestion the next time. 
I also just noticed, that this PR already contains the "drop index" changes I proposed with approved #9112.

File: chia/wallet/trading/trade_store.py [hunk 10]
```diff
@@ -43,7 +43,11 @@ async def create(cls, db_wrapper: DBWrapper, cache_size: uint32 = uint32(600000)
             "CREATE INDEX IF NOT EXISTS trade_confirmed_index on trade_records(confirmed_at_index)"
         )
         await self.db_connection.execute("CREATE INDEX IF NOT EXISTS trade_status on trade_records(status)")
-        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS trade_id on trade_records(trade_id)")
+
+        # earlier versions of chia created this index despite no lookups needing
```
Comment (Yostra) [inline chia/wallet/trading/trade_store.py:47]: this was "un-released" feature

**Comment by Yostra** [comment]
Let's keep this open and unmerged. This won't get into the release prior to merging standalone wallet (next release most likely), and that changes db quite a bit + uses a different file. Let's cherry pick then relevant changes in.

**Comment by Yostra** [comment]
plust trades are WiP.

**Comment by neurosis69** [comment]
Mostly obsolete, the remaining wallet index issues were showed in issue #10276
